src/services/MapService.js

```diff
@@ -127,4 +130,15 @@ export class MapService {
 		const element = document.querySelector('ba-footer')?.shadowRoot.querySelector('.scale');
 		return element ?? null;
 	}
+
+	/**
+	 * Returns a viewport rectangle, where the map element is not overlapped by any other relevant element
```

Comment:
Returns a DOMRect instance describing the visible part of the map (which means the part of the part that is not overlapped by any other UI element)